thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/views/AnalysisView.js

```diff
@@ -8,7 +8,8 @@ function AnalysisView(analysisModel) {
 }
 
 AnalysisView.prototype = {
-  init: function () {
+  init({metricId=""}) {
```

Comment:
In this case, `init` must be called with `{}` and will throw an error in `init()` case.
I suggest to use like
```
init(params = {}) {
  const {metricId} = params;
  this.viewParams.metricId = metricId || '';
}
```